llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2989,7 +2989,14 @@ InstructionCost AArch64TTIImpl::getGatherScatterOpCost(
       ElementCount::getScalable(1))
```

Comment:
If we bail out when `isLegalMaskedGather` returns false, which indeed would happen with such types, – as per @alexey-bataev suggestion above – I don't think we would ever be reaching this point.

Reply:
It didn't look like that function was called when I tried it, and it is best if these functions work correctly in isolation. It's possible to call it throught the cost model: https://godbolt.org/z/z4635vodq.

I think that because the gather will not become a scalable SVE gather in this case, the line above with useNeonVector should be returning true so we use the BaseT::getGatherScatterOpCost for the cost.